### ISSUE TITLE
Metadata: Can't remove trailing slash from canonical url when using metadata base. 

### ISSUE THREAD

**Issue by lucashfreitas** [vercel/next.js#54070]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
perating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.16.0
      npm: 9.5.1
      Yarn: 1.22.17
      pnpm: 8.6.9
    Relevant Packages:
      next: 13.4.16-canary.1
      eslint-config-next: 13.4.16
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router, Metadata (metadata, generateMetadata, next/head)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/lucashfreitas/nextjs-canonical-url-trailing-slash

### To Reproduce

I'd like to have a canonical url for the base URL which does not have a trailing slash, but it's not possible to do it when using metadatabase.  

1. Clone the repository. (https://github.com/lucashfreitas/nextjs-canonical-url-trailing-slash)

2. Run `next dev`

3. Note that the head `<link rel="canonical"` has the baseURL followed by a trailing slash. 

4. There is no option to render the canonical URL for the base URL without the trailing slash using metdatabase: 
- If we try to use `''` or `null` for the canonical URL, nextjs won't render the `<link>` to the head.

### Describe the Bug

When using metadatabase and canonical URL is not possible to remove the trailing slash from the base URL. According to the documentation and example available here https://nextjs.org/docs/app/api-reference/functions/generate-metadata, if we set the canonical URL to `/` , nextjs would render only the metadataBase, but it's actually adding a trailing slash to the end. 

- when canonical is `/`: nextjs adds the trailing slash
- when canonical is null or '': nextjs does not add a canonical link to the head.
- when canonical URL is the base URL e.g `https://acme.com`: nextjs renders the baseUrl followed by a trailing slash.

<img width="837" alt="image" src="https://github.com/vercel/next.js/assets/11416392/bbeddfd7-9ef2-4c05-a718-0db3ab7d9331">

### Expected Behavior

I'd expect metadataBase to not add a trailing slash for the root/base URL. If my website is https://acme.com , I'd expect a `<link rel="canonical" href="https://acmes.com">` without the trailing slash, but it does not seems possible when using metadatabase. (I verified that changing the trailing slash setting on `next.config.js` also doesn't work).

According to the example mentioned here https://nextjs.org/docs/app/api-reference/functions/generate-metadata it should work but it doesn't seems to. 

<sub>[NEXT-2424](https://linear.app/vercel/issue/NEXT-2424/metadata-cant-remove-trailing-slash-from-canonical-url-when-using)</sub>

**Comment by RobSchilderr** [comment]
Same issue here. Also when adding the alternates at a specific page in the `app` router:
``` 
alternates: {
    canonical: 'https://domain.app/privacy',
    languages: {
      'en-US': '/en-US',
      'pl-PL': '/pl-PL',
      'tr-TR': '/tr-TR',
    },
  }
```
  It always adds the language at the end, like: `https://domain.app/privacy/en-US`, while packages like `next-intl` and `next-translate`  work with `https://domain.app/en-US/privacy`.

**Comment by DennieMello** [comment]
I encountered the same error

**Comment by danoph** [comment]
Same here.

**Comment by alessiofrittoli** [comment]
You can omit the `metadatabase` property but you have to prefix opengraph urls with your site origin manually.
You'll get a notice in console eventhough.

This has no sense at all since Next.js should do it for you but I faced the same issue and stick to this solution for now.

**Comment by bramvdpluijm** [comment]
I'm also encountering this issue. Any workarounds yet?

**Comment by Hugomndez** [comment]
Currently, I am omitting the ```Metadata.metadataBase``` while waiting for the fix.

**Comment by bh1337x** [comment]
This may not be a Next.js issue. The URL object in both Browser and Node.js act this way.
![image](https://github.com/vercel/next.js/assets/22233888/35403dac-1cf8-444a-bff7-68acec7738ea)
![image](https://github.com/vercel/next.js/assets/22233888/945869a1-4d61-46e5-afc7-700adadb1604)

**Comment by Maxservais** [comment]
Also facing the same issue and as a result GSC complains as the canonical url (that includes a trailing slash) is different from the one in my sitemap

**Comment by paschalidi** [comment]
Have been seeing the same, thank you for fixing it. Would it be right to assume that this fix will be released on the `14.1.1`?

**Comment by huozhi** [comment]
The fix is landed in 14.1.1!
